nsq/reader.py

```diff
@@ -665,10 +646,21 @@ def _redistribute_rdy_state(self):
                     logger.info('[%s:%s] idle connection, giving up RDY count', conn.id, self.name)
                     self._send_rdy(conn, 0)
 
+            conns = self.conns.values()
+
+            in_flight = [c for c in conns if c.in_flight]
```

Comment:
I still don't quite understand why we need to use `in_flight` when `total_rdy` should represent the same thing? The accounting for both `total_rdy` and `in_flight` used to happen [in 2 subsequent lines in the same block](https://github.com/nsqio/pynsq/pull/179/files#diff-9fd12dcd9000f610741f4d399c31e1b8R489)).